src/main/java/org/opensearch/ad/transport/ADStatsNodeRequest.java

```diff
@@ -13,14 +13,15 @@
 
 import java.io.IOException;
 
+import org.opensearch.action.ActionRequest;
+import org.opensearch.action.ActionRequestValidationException;
 import org.opensearch.common.io.stream.StreamInput;
 import org.opensearch.common.io.stream.StreamOutput;
-import org.opensearch.transport.TransportRequest;
 
 /**
  *  ADStatsNodeRequest to get a nodes stat
  */
-public class ADStatsNodeRequest extends TransportRequest {
+public class ADStatsNodeRequest extends ActionRequest {
```

Comment:
Why are we changing the request here?

Reply:
Because as per the changes made by dan in TransportRequest, it now accepts the request which extends the ActionRequest and response which extends ActionResponse

Reply:
`ActionRequest` extends `TransportRequest` so this doesn't break anything but does add some functionality relying on `ActionRequest` signature. Is this being triggered remotely from JS, etc. and that's why we need the `ActionRequest`?

Reply:
No this request is not being triggered from JS

Reply:
Same question as Owais, if its not necessary to change then lets revert this change

Reply:
As suggested by @dbwiddis I will open a new issue where we will cater the blast radius of TransportNodesAction and will do the required work there.

Reply:
Did you get any error if you keep the request as `TransportRequest`?

Reply:
yes I got an error, TransportAction only accepts the requests which extends ActionRequest.